pandas/core/format.py

```diff
@@ -2101,6 +2105,32 @@ def _format_strings(self):
 
         return fmt_values
 
+    def get_formatted_data(self):
+        values = self.values
```

Comment:
add a doc-string here describing what this is doing. Some comments in the code as well (to describe the purposes of the if clauses)